lib/floe/workflow/states/task.rb

```diff
@@ -82,54 +87,6 @@ def success?(context)
           runner.success?(context.state["RunnerContext"])
         end
 
-        def find_retrier(error)
-          self.retry.detect { |r| r.match_error?(error) }
-        end
-
-        def find_catcher(error)
-          self.catch.detect { |c| c.match_error?(error) }
-        end
-
-        def retry_state!(context, error)
-          retrier = find_retrier(error["Error"]) if error
-          return if retrier.nil?
-
-          # If a different retrier is hit reset the context
-          if !context["State"].key?("RetryCount") || context["State"]["Retrier"] != retrier.error_equals
-            context["State"]["RetryCount"] = 0
-            context["State"]["Retrier"]    = retrier.error_equals
-          end
-
-          context["State"]["RetryCount"] += 1
-
-          return if context["State"]["RetryCount"] > retrier.max_attempts
-
```

Comment:
Always wanted the retrier / catcher to act just like a `State`.

You ask - do you have a retrier/catcher for me?
And if you do, then you `run_nonblock` it just like you do for the current state.

Reply:
ignore ^ - I can play with this later